File: src/main/java/com/senderman/lastkatkabot/command/user/PayRespects.java
package com.senderman.lastkatkabot.command.user;

import com.annimon.tgbotsmodule.api.methods.Methods;
import com.annimon.tgbotsmodule.services.CommonAbsSender;
import com.senderman.lastkatkabot.callback.Callbacks;
import com.senderman.lastkatkabot.command.CommandExecutor;
import com.senderman.lastkatkabot.util.callback.ButtonBuilder;
import com.senderman.lastkatkabot.util.callback.MarkupBuilder;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

@Component
public class PayRespects implements CommandExecutor {

    private final CommonAbsSender telegram;

    public PayRespects(CommonAbsSender telegram) {
        this.telegram = telegram;
    }

    @Override
    public String getTrigger() {
        return "/f";
    }

    @Override
    public String getDescription() {
        return "(reply) press f to pay respects. А можно вот так: /f штаны за 40 хривень. Или просто /f";
    }

    @Override
    public void execute(Message message) {

        String object;

        if (message.getText().split("\\s+").length > 1) { // when object defined in the message text
            object = "to " + message.getText().split("\\s+", 2)[1];
        } else if (message.isReply()) { // when object is another user
            object = "to " + message.getReplyToMessage().getFrom().getFirstName();
        } else {
            object = "";
        }

        Methods.deleteMessage(message.getChatId(), message.getMessageId()).callAsync(telegram);
        var text = "🕯 Press F to pay respects " + object +
                   "\n" + message.getFrom().getFirstName() + " has paid respects";

        var markup = new MarkupBuilder()
                .addButton(ButtonBuilder.callbackButton()
                        .text("F")
                        .payload(Callbacks.F))
                .build();

        Methods.sendMessage(message.getChatId(), text)
                .setReplyMarkup(markup)
                .callAsync(telegram);
    }
}

File: src/main/kotlin/com/senderman/lastkatkabot/usercommands/PayRespects.kt
package com.senderman.lastkatkabot.usercommands

import com.annimon.tgbotsmodule.api.methods.Methods
import com.senderman.lastkatkabot.LastkatkaBotHandler
import com.senderman.lastkatkabot.callbacks.Callbacks
import com.senderman.neblib.CommandExecutor
import org.telegram.telegrambots.meta.api.objects.Message
import org.telegram.telegrambots.meta.api.objects.replykeyboard.InlineKeyboardMarkup
import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.InlineKeyboardButton

class PayRespects(private val handler: LastkatkaBotHandler) : CommandExecutor {
    override val command: String
        get() = "/f"
    override val desc: String
        get() = "(reply) press f to pay respects. А можно вот так: /f штаны за 40 хривень. Или просто /f"

    override fun execute(message: Message) {
        if (message.isUserMessage) return
        if (message.isReply && message.from.firstName == message.replyToMessage.from.firstName) return

        val obj: String = when {
            message.text.split(" ").size > 1 -> "to " + message.text.split(" ", limit = 2)[1]
            message.isReply -> "to " + message.replyToMessage.from.firstName
            else -> ""
        }

        Methods.deleteMessage(message.chatId, message.messageId).call(handler)
        val text = "\uD83D\uDD6F Press F to pay respects $obj" +
                "\n${message.from.firstName} has payed respects"
        handler.sendMessage(
            Methods.sendMessage()
                .setChatId(message.chatId)
                .setText(text)
                .setReplyMarkup(markupForPayingRespects)
        )
    }

    private val markupForPayingRespects: InlineKeyboardMarkup
        get() {
            val markup = InlineKeyboardMarkup()
            markup.keyboard = listOf(
                listOf(
                    InlineKeyboardButton()
                        .setText("F")
                        .setCallbackData(Callbacks.PAY_RESPECTS)
                )
            )
            return markup
        }
}